src/LandmarkParser.java

```diff
@@ -0,0 +1,38 @@
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+
+public class LandmarkParser {
+
+
+  public static HashMap<Integer, Landmark> parseLandmarks() {
+    HashMap<Integer, Landmark> list = new HashMap<>();
```

Comment:
Wasn't the goal to use a linked list? It seems that you used a hash map instead. A linked list in this situation would work very nicely. In the `Landmark` class, there should be a pointer to the next landmarks.

Reply:
I changed my mind about using a linked list.  I think a hashmap works just fine.